pkg/armhelpers/kubeclient.go

```diff
@@ -0,0 +1,143 @@
+package armhelpers
+
+import (
+	"time"
+
+	log "github.com/sirupsen/logrus"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/fields"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/pkg/api/v1"
+	policy "k8s.io/client-go/pkg/apis/policy/v1beta1"
+	"k8s.io/client-go/tools/clientcmd"
+	clientcmdapi "k8s.io/client-go/tools/clientcmd/api"
+)
+
+const (
+	evictionKind        = "Eviction"
+	evictionSubresource = "pods/eviction"
+)
+
+//KubernetesClientSetClient kubernetes client hooked up to a live api server
+type KubernetesClientSetClient struct {
+	clientset         *kubernetes.Clientset
+	interval, timeout time.Duration
+}
+
+//GetKubernetesClient returns a KubernetesClient hooked up to the api server at the masterURL
+func (az *AzureClient) GetKubernetesClient(masterURL, kubeConfig string, interval, timeout time.Duration) (KubernetesClient, error) {
+	// creates the clientset
+	config, err := clientcmd.BuildConfigFromKubeconfigGetter(masterURL, func() (*clientcmdapi.Config, error) { return clientcmd.Load([]byte(kubeConfig)) })
+	if err != nil {
+		return nil, err
+	}
+	clientset, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		return nil, err
+	}
+	return &KubernetesClientSetClient{clientset: clientset, interval: interval, timeout: timeout}, nil
+}
+
+//ListPods returns all Pods running on the passed in node
+func (c *KubernetesClientSetClient) ListPods(node *v1.Node) (*v1.PodList, error) {
+	return c.clientset.Core().Pods(metav1.NamespaceAll).List(metav1.ListOptions{
+		FieldSelector: fields.SelectorFromSet(fields.Set{"spec.nodeName": node.Name}).String()})
+}
+
+//GetNode returns details about node with passed in name
+func (c *KubernetesClientSetClient) GetNode(name string) (*v1.Node, error) {
+	return c.clientset.Nodes().Get(name, metav1.GetOptions{})
+}
+
+//UpdateNode updates the node in the api server with the passed in info
+func (c *KubernetesClientSetClient) UpdateNode(node *v1.Node) (*v1.Node, error) {
+	return c.clientset.Nodes().Update(node)
+}
+
+//SupportEviction queries the api server to discover if it supports eviction, and returns supported type if it is supported
+func (c *KubernetesClientSetClient) SupportEviction() (string, error) {
```

Comment:
Should some of this be written upstream? Advantages I can think of is that bugs fixes could be done by the community. We don't need to keep up with new features in the client. If we went the new client then `SupportEviction` comes with it. Otherwise with each new version of client this code could fall behind.